### ISSUE TITLE
Data hosting and distribution

### ISSUE THREAD

**Issue by answerquest** [osrm-decentralized/osrm_IN_docker#5]
## Problem : Docker LFS has v.low free limit
I did a download of the files from git LFS yesterday evening, and got this alert from docker just now:

```
Sub: GitHub] Git LFS disabled for osrm-decentralized
Git LFS has been disabled on the organization osrm-decentralized because you’ve exceeded your data plan by at least 150%. Please purchase additional data packs to cover your bandwidth and storage usage:

  https://github.com/organizations/osrm-decentralized/billing/data/upgrade

Current usage as of 03 May 2020 03:21PM UTC:

  Bandwidth: 0.0 GB / 1 GB (0%)
  Storage: 7.38 GB / 1 GB (738%)
```
The prices on the page are just silly - not worth it.

I'd like to suggest using our own LFS instead - and we can use a new server I've started up for initial trial if needed.

## The data in question
```
                                                                                           
    1.5 GiB [######### ]  india-latest.osrm.cell_metrics
  728.3 MiB [####      ]  india-latest.osrm.geometry
  701.4 MiB [####      ]  india-latest.osrm.fileIndex
  611.6 MiB [###       ]  india-latest.osrm.mldgr
  595.2 MiB [###       ]  india-latest.osrm.ebg
  407.0 MiB [##        ]  india-latest.osrm.nbg_nodes
  297.6 MiB [#         ]  india-latest.osrm.turn_penalties_index
  173.6 MiB [#         ]  india-latest.osrm.edges
  147.7 MiB [          ]  india-latest.osrm.ebg_nodes
  147.2 MiB [          ]  india-latest.osrm.enw
  134.5 MiB [          ]  india-latest.osrm.icd
  100.6 MiB [          ]  india-latest.osrm.cnbg_to_ebg
  100.6 MiB [          ]  india-latest.osrm.cnbg
   98.2 MiB [          ]  india-latest.osrm.partition
   49.6 MiB [          ]  india-latest.osrm.turn_weight_penalties
   49.6 MiB [          ]  india-latest.osrm.turn_duration_penalties
   10.9 MiB [          ]  india-latest.osrm.cells
    2.8 MiB [          ]  india-latest.osrm.ramIndex
    1.2 MiB [          ]  india-latest.osrm.names
   68.0 KiB [          ]  india-latest.osrm.datasource_names
    8.0 KiB [          ]  india-latest.osrm.tls
    8.0 KiB [          ]  india-latest.osrm.properties
    8.0 KiB [          ]  india-latest.osrm.maneuver_overrides
    4.0 KiB [          ]  india-latest.osrm.tld
    4.0 KiB [          ]  india-latest.osrm.restrictions
    4.0 KiB [          ]  india-latest.osrm.timestamp
```

### 1. Direct dowload:
- have the data/ folder mounted as persistent storage volume
- Host all files on an accessible URL path, and add `wget` lines in dockerfile
- Use `wget -N` to only download if there is a newer version on server, if the downloaded files already exist on the persistent volume. This ensures that restarting the deployment won't trigger unnecessary fresh downloads.

### 2. Compressed:
If we want to do some bandwidth saving:
- have the data/ folder mounted as persistent storage volume
- compress into single or multi-part : https://superuser.com/a/184601/487360
- host on server
- in Dockerfile, download using `wget -N`
- then add uncompress command in Dockerfile

A related solution where they combined the wget and uncompress and cleanup commands: https://stackoverflow.com/a/45987109/4355695

### 3. Torrent ?!?
Just for brainstorming: Here's another way that I think can scale up if this ever gets famous : replicate how there are torrent options for downloading linux images. This can be for just the data, or this whole repo including data files could just be seeded out over torrent from one starting point (always-on server - We can use one of mine for pilot), and we could reach out to other folks to co-seed.
- https://www.addictivetips.com/ubuntu-linux-tips/set-up-transmission-bittorrent-server-on-linux/
- https://lifehacker.com/how-to-share-your-own-files-using-bittorrent-5534190

One problem I foresee with this is in updating the data if we want to re-build using latest osm data at say monthly interval. But then for folks really needing that, maybe we can just have a simpler alternative repo where just the .pbf is downloaded and the user has to do the building of osrm graph by themselves (and have larger RAM available for the same).

**Comment by DaxServer** [comment]
One possibility is to host the files on our server and serve them ourselves. Here is an exhaustive list of open source file sharing platforms - https://github.com/awesome-selfhosted/awesome-selfhosted#file-sharing-and-synchronization

Nextcloud and ownCloud seem promising.

**Comment by answerquest** [comment]
Yes. One can also just install Apache or Nginx on their server and locate the files somewhere under /var/www/html/ (apache) , /usr/share/nginx/html (nginx). That's what I've done here: https://server.nikhilvj.co.in/dump/ and there is a shell script at back regenerating those files from fresh osm data every day.

**Comment by DaxServer** [comment]
I am just thinking too complex, ain't I? I will install the Nginx and enable directory listing and that's it. Maybe also a shell script which will download all the 10+ files instead of doing 10 `wget`s.

**Comment by DaxServer** [comment]
I have merged #6 to remove LFS as it is not a sustainable solution. Let's use the file serving using Nginx
